components/scream/src/physics/p3/p3_functions_check_values_impl.hpp

```diff
@@ -0,0 +1,84 @@
+
+#ifndef P3_FUNCTIONS_CHECK_VALUES_IMPL_HPP
+#define P3_FUNCTIONS_CHECK_VALUES_IMPL_HPP
+
+#include "p3_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace p3 {
+
+/*!------------------------------------------------------------------------------------
+  Checks current values of prognotic variables for reasonable values and
+  stops and prints values if they are out of specified allowable ranges.
+  
+  'check_consistency' means include trap for inconsistency in moments;
+  otherwise, only trap for Q, T, and negative Qx, etc.  This option is here
+  to allow for Q<qsmall.and.N>nsmall or Q>qsmall.and.N<small which can be produced
+  at the leading edges due to sedimentation and whose values are accpetable
+  since lambda limiters are later imposed after SEDI (so one does not necessarily
+  want to trap for inconsistency after sedimentation has been called).
+  
+  The value 'source_ind' indicates the approximate location in 'p3_main'
+  from where 'check_values' was called before it resulted in a trap.
+  -----------------------------------------------------------------------------------
+*/
+template <typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::check_values(const view_1d<Spack>& qv, const view_1d<Spack>& temp, const Int& kts, const Int& kte, 
+               const Int& timestepcount, const bool& force_abort, const Int& source_ind, const MemberType& team, 
+               const view_1d<Spack>& col_loc, view_1d<Smask> qv_out_bounds, view_1d<Smask> t_out_bounds)
```

Comment:
I think qv_out_bounds and t_out_bounds are for testing only. In practice we don't want them. So rewrite this routine to accept them for testing but not otherwise. You could use `view_1d<Smask>* qv_out_bounds = nullptr` for example and assign only if not null.